sentry/gatsby doesn't support Gatsby 4 as a peer dependency

## Package + Version

- [X] other: @sentry/gatsby

### Version:

```
6.13.3
```

## Description

@sentry/gatsby has a peer dependency on gatsby@"^2.0.0 || ^3.0.0" but ^4.0.0 is out

To repro, setup a package.json with just latest gatsby and @sentry/gatsby, using node 16 or later. Try `npm install` without running --force or --legacy-peer-deps; it errors with:

```npm ERR! Found: gatsby@4.0.2
npm ERR! node_modules/gatsby
npm ERR!   gatsby@"^4.0.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer gatsby@"^2.0.0 || ^3.0.0" from @sentry/gatsby@6.13.3
npm ERR! node_modules/@sentry/gatsby
npm ERR!   @sentry/gatsby@"^6.13.3" from the root project```

--- a/packages/gatsby/package.json
+++ b/packages/gatsby/package.json
@@ -30,7 +30,7 @@
     "@sentry/tracing": "6.14.0"
   },
   "peerDependencies": {
-    "gatsby": "^2.0.0 || ^3.0.0"
+    "gatsby": "^2.0.0 || ^3.0.0 || ^4.0.0"
   },
   "devDependencies": {
     "@sentry/types": "6.14.0",

